components/dialog/dialog-mixin.js

```diff
@@ -177,10 +177,33 @@ export const DialogMixin = superclass => class extends RtlMixin(superclass) {
 		}
 	}
 
+	_findAutofocusElement(node) {
+		if (this._useNative) {
+			// Do not override native autofocus attribute implementation
+			return null;
+		}
+
+		if (!node.querySelector('slot')) {
```

Comment:
I think I would optimize this a little to only get the slot once.

``` javascript
const slot = node.querySelector('slot');
if (!slot) return null;

const content = slot.assignedElements({ flatten: true });
...
```

I think it should be possible to use `el.hasAttribute('autofocus')`.

Reply:
Wouldn't `hasAttribute` give a false positive if we get something like `<p autofocus="false">`?

Reply:
That's not a false positive -- the presence or absence of boolean attributes make them true or false, not the attribute value. `autofocus="false"` is the same as `autofocus` is the same as `autofocus="yes please"` -- the attribute value doesn't matter.

Reply:
Fixed with https://github.com/BrightspaceUI/core/pull/2618/commits/1b91e5b1b0dcf9ef7eb2bcf1711a5f2378e2f53a